### DISCUSSION TITLE
TimePickerThemeMinWidth doesn't update view

### DISCUSSION THREAD

**Discussion by BullT** [unoplatform/uno#7806] [Q&A]
### Current behavior

HI,

I want change the width of TimePicker.
To do that I add this line in the App.xaml:

```
<x:Double x:Key="TimePickerThemeMinWidth">140</x:Double>
```

Like this:
```
<Application
    x:Class="unoappTimePickerWidth.App"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:unoappTimePickerWidth">
	
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
                <!-- Place resources here -->
            </ResourceDictionary.MergedDictionaries>
            <!-- During the debugin comment and uncoment the next line -->
            <x:Double x:Key="TimePickerThemeMinWidth">140</x:Double>
        </ResourceDictionary>
    </Application.Resources>

</Application>
```

But it doesn't work.
To do that it works, during the debuging I have to comment and uncoment that line.

And this is the MainPage.xaml:
```
<Page
    x:Class="unoappTimePickerWidth.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:unoappTimePickerWidth"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
        <TimePicker HorizontalAlignment="Center" ClockIdentifier="24HourClock"/>
    </Grid>
</Page>
```

### Workaround

Write de basic program with TimePiker in MainPage.xaml and add the <x:Double x:Key="TimePickerThemeMinWidth">140</x:Double> to App.xaml.
Run in debug mode and comment and uncomment this line <x:Double x:Key="TimePickerThemeMinWidth">140</x:Double> in App.xaml to see the TimePicker width change.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

3.1.2

### Affected platforms

Windows

### IDE

Visual Studio 2022

### IDE version

17.0.4

**Comment by jeromelaban** [comment]
Thanks for the report. You're reporting this issue as a Windows target, which means that Uno cannot fix the issue. I'm going to transfer this as a discussion instead.

Can you provide a sample app with repro steps?

**Reply by BullT** [reply]
Hi Jerome,

This is the link to GitHub:

https://github.com/BullT/AppUnoTimePicker.git

Run the App in debug mode.
Coment the line ```<x:Double x:Key="TimePickerThemeMinWidth">140</x:Double>``` in App.xaml.
Go to App.
Uncommnet the ```<x:Double x:Key="TimePickerThemeMinWidth">140</x:Double>``` in App.xaml.
Go to App and now the width change.

Thanks,
BullT

**Reply by jeromelaban** [reply]
To be clear, the which head are you running for the repro?

**Reply by BullT** [reply]
Sorry, UWP

**Reply by jeromelaban** [reply]
Thanks. If it does not work in UWP, it's not something we can change. What you'll want to do for this is close the complete style of the time picker and adjust the values there.

**Reply by BullT** [reply]
Thank Jerome,

I'm trying to do it but doesn't work maybe I don't know how to do it.
I've looked in Google but no answer.

Can you tell me how to do it?

Best regards,
BullT

**Reply by BullT** [reply]
Hi,

I found the solution:

https://docs.microsoft.com/en-us/answers/questions/698482/timepickerthememinwidth-doesn39t-update-view.html
https://stackoverflow.com/questions/31918302/modifying-default-style-of-timepicker-and-datepicker-controls

Thanks
